kinto/core/resource/__init__.py

```diff
@@ -245,6 +245,12 @@ def collection_get(self):
         self._raise_412_if_modified()
 
         headers = self.request.response.headers
+
+        if getattr(self.context, "forced_empty_list", False):
+            # Return an empty list.
+            headers['Total-Records'] = encode_header('0')
+            return self.postprocess([])
+
```

Comment:
I would have preferred to keep this close to the existing (and only) link between context and resource : https://github.com/Kinto/kinto/blob/master/kinto/core/resource/__init__.py#L1122-L1135
For example, extract_filters could return something like (last_modified == -1) to return 0 records.